assets/src/components/CustomLink/CustomLink.js

```diff
@@ -0,0 +1,48 @@
+import {Component} from '@wordpress/element';
+import {URLInput} from '@wordpress/block-editor'
+import {TextControl as BaseTextControl} from '@wordpress/components';
+import withCharacterCounter from '../../components/withCharacterCounter/withCharacterCounter';
+
+const TextControl = withCharacterCounter( BaseTextControl );
+
+export class CustomLink extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+
+  isCustomLinkValid() {
+    return !!this.props.value ? this.props.value.substr(0, 8).toLowerCase() === 'https://' : true;
```

Comment:
Maybe it can be a bit nicer if we use javascript's API for URLs? See this SO answer: https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url/49849482#answer-43467144

Maybe it's also more readable if you make a separate return statement before this one, if `this.props.value` is false. There's already a lot of things going on on this line :)

Reply:
I split up the falsy check with the validation check.